pycbc/inference/burn_in.py

```diff
@@ -415,13 +476,23 @@ def nacl(self, filename):
         """Carries out the nacl test and stores the results."""
         pass
 
+    def chain_moments(self):
+        """Carries out the chain_moments test and stores the results. """
+        pass
+
     @abstractmethod
     def evaluate(self, filename):
-        """Performs all tests and evaluates the results to determine if and
-        when all tests pass.
+        """Performs all tests which are performed on checkpoint files and 
+        evaluates the results to determine if and when all tests pass.
         """
         pass
 
+    def evaluate_sampler(self):                                                
+        """Performs all tests which are performed on sampler and evaluates 
+        the results to determine if and when all tests pass.                                                     
+        """                                                                      
+        pass
```

Comment:
Don't pass here, just move the function that you've added to the class below here, since it'll work as-is for all types of MCMC samplers.